Prevent urlencode throw an exception

tiran133 opened the pull request:
`declare(strict_types=1);`
Introduced here
https://github.com/CatoTH/antragsgruen/commit/349522e7996c46ccb9007ac00c191b29aee8df7b#diff-5bd33395e5715c82071cef80cbfbb57ff327fd27e99c62c6d39da556d83c074bR3

This change makes no authentication throw, an exception

We use IP based auth and don't; need username and password. 

`urlencode` expects a string. Before strict_types it was ok to pass null now it throws.

CatoTH commented:
Hi!
Oh boy, you're right. Sorry!
But while we're at it, would the DSN work in your case if it was just
```
$dsn = 'smtp://' . urlencode($this->host) . ':' . $this->port;
```
? That is, just skipping the username/password part entirely?
Because I think, I would prefer leaving username/passwort to be default null, and then asseble the DSN accordingly.

tiran133 commented:
Hi,

Yes, it works the same way. I tested it on our staging environment.
I reverted the change and added an if check for the dns string.

Let me know if that is ok or if you preferrer ternary operator like this

```php
$dsn = $this->username && $this->password ?
                'smtp://' . urlencode($this->username) . ':' . urlencode($this->password) . '@' .
                urlencode($this->host) . ':' . $this->port
                : 'smtp://' . urlencode($this->host) . ':' . $this->port;
```

I think in this case a proper "if", is more readable.

CatoTH commented:
Hi!
Thank you! And yes, I agree about if/else :-)
I suppose you do have the means to fix it on your installation without a tagged release? I will include the fix in the next bugfix release, but I'm typically collecting fixes for a bit before creating a new release.

tiran133 commented:
Thanks for merging. 

All good, next release will be fine. 

I have fixed it in my installation.
